examples/docker.yaml

```diff
@@ -45,7 +45,20 @@ provision:
     set -eux -o pipefail
     command -v docker >/dev/null 2>&1 && exit 0
     export DEBIAN_FRONTEND=noninteractive
-    curl -fsSL https://get.docker.com | sh
+
+    # DOCKER_INSTALL_START
+    # From official Docker Engine installation guide: https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository
+    apt-get -y install ca-certificates curl gnupg lsb-release
+
+    mkdir -p /etc/apt/keyrings
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
```

Comment:
Why tee?

Reply:
None of these commands are brought up by me, rather they're from [here](https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script).

Reply:
Tee is used upstream because of it's usage with `sudo`, because `sudo echo '...' > output` only applies sudo to the `echo`, not the file direct `>`.